preprocessing/structs.ts

```diff
@@ -1,15 +1,82 @@
 /* # Mangrove Summary
-   * The Mangrove holds order books for `outbound_tkn`,`inbound_tkn` pairs.
-   * Offers are sorted in a doubly linked list.
+   * Mangrove holds offer lists for `outbound_tkn`,`inbound_tkn` pairs with a given `tickSpacing`.
+   * Offers are sorted in a tree (the "tick tree") where each available price point (a `bin`) holds a doubly linked list of offers.
    * Each offer promises `outbound_tkn` and requests `inbound_tkn`.
    * Each offer has an attached `maker` address.
-   * In the normal operation mode (called Mangrove for Maker Mangrove), when an offer is executed, we:
+   * When an offer is executed, Mangrove does the following:
      1. Flashloan some `inbound_tkn` to the offer's `maker`.
      2. Call an arbitrary `execute` function on that address.
      3. Transfer back some `outbound_tkn`.
      4. Call back the `maker` so they can update their offers.
     
    **Let the devs know about any error, typo etc, by contacting 	devs@mangrove.exchange**
+
+
+  There is one Mangrove contract that manages all tradeable offerLists. This reduces deployment costs for new offerLists and lets market makers have all their provision for all offerLists in the same place.
+
+   The interaction map between the different actors is as follows:
+   <img src="./contactMap.png" width="190%"></img>
```

Comment:
The contactMap.png is outdated. It doesn't seem to be a DrawIO file, so not sure how to update it?

![billede](https://github.com/mangrovedao/mangrove-core/assets/3458856/c51d0ee4-eaf7-436a-9e2f-8938b39573f6)

Reply:
I think this is a omnigraffle file, Jean has the sources. I think we could leave it for now and redo this one and the others with a proper Mangrove style later, what do you think?

Reply:
alright. Do we have a backlog somewhere for things to fix post code freeze?

Reply:
No, I'll add it to the ticks PR.